### DISCUSSION TITLE
Migration from CNI to netavark

### DISCUSSION THREAD

**Discussion by dcermak** [containers/podman#20919] [Ideas]
The CNI networking backend will be deprecated with the upcoming podman 5.0.0 release and netavark is already the new default. Are there any plans for a migration from CNI to netavark beyond tear down everything and recreate? If yes, can we contribute ideas/code in any way?

**Comment by Luap99** [comment]
Assuming users only use standard CNI config files (ones that were created by `podman network create`) then a migration is at least in theory possible. However we do not have any plans to do this automatically as there is a lot that can go wrong.

That said I have nothing against documenting steps to migrate without a full blown `podman system reset`.
The easiest is if users do not have any networks beside the default podman bridge network. If there are other networks then you need to migrate the files into the netavark format. The good thing here is that podman can already do this: use `podman network inspect <name> | jq .[]  > $path` where $path is `/etc/containers/networks/<name>.json` for root and `~/.local/share/containers/storage/networks/<name>.json` for rootless. Unless you changed the storage paths via other config options of course. But again this would only work for files that were created with podman network create, custom CNI config files would not be supported and likely create broken config files.

Now stop all containers and then make sure to delete any leftover CNI firewall rules and CNI bridge interfaces (reboot is likely easier) then update to 5.0 or switch the backend in containers.conf to netavark and then start the containers again and it should keep working.
